ibis/backends/tests/tpc/ds/test_queries.py

```diff
@@ -942,7 +957,8 @@ def test_12(web_sales, item, date_dim):
         .mutate(
             revenueratio=_.itemrevenue
             * 100.000
-            / _.itemrevenue.sum().over(group_by=_.i_class)
+            # snowflake divide by zero is an error, all others return a value (null, nan, or inf)
```

Comment:
Do we want to unify this behavior?

I think there's only one option given the support matrix for nan and infinity looks like Swiss cheese, and that's to return `NULL` on divide by zero.

Errors are out, because they'd require looking at the value (or using some kind of database error function, which is also not well-supported).

Reply:
Maybe an API that's explicit about this would be better, like `t.x.safe_div(t.y)` naming TBD…

Reply:
Hm, I think this might be a window `range` issue as well. Looking…

Reply:
Nope, divide by zero still, even when being explicit about the `range=(None, 0)` bits.

Reply:
Hilarious, sqlglot is inserting a `NULLIF(..., 0)` during the transpile:

```
(Pdb) sg.parse_one('select x / y', read='duckdb').sql('snowflake')
'SELECT x / NULLIF(y, 0)'
```

Reply:
Hm, wonder why we aren't picking that up in the snowflake compiler. Looking into it.

Reply:
I think returning NULL for a divide-by-zero is reasonable?

Reply:
Ah, because sqlglot has knowledge of what the origin dialect was, so it knows that to emulate the same behavior it has to make the `Div` operator as `safe=True` 🔍